tests/unit/scripting.tcl

```diff
@@ -1404,71 +1404,117 @@ start_server {tags {"scripting"}} {
 # Additional eval only tests
 start_server {tags {"scripting"}} {
     test "Consistent eval error reporting" {
+        r config resetstat
         r config set maxmemory 1
         # Script aborted due to Redis state (OOM) should report script execution error with detailed internal error
-        assert_error {ERR Error running script (call to *): @user_script:*: OOM command not allowed when used memory > 'maxmemory'.} {
+        assert_error {OOM command not allowed when used memory > 'maxmemory'*} {
             r eval {return redis.call('set','x','y')} 1 x
         }
+        assert_equal [errorrstat OOM r] {count=1}
+
         # redis.pcall() failure due to Redis state (OOM) returns lua error table with Redis error message without '-' prefix
+        r config resetstat
         assert_equal [
             r eval {
                 local t = redis.pcall('set','x','y')
-                if t['err'] == "OOM command not allowed when used memory > 'maxmemory'." then
+                if t['err'] == "command not allowed when used memory > 'maxmemory'." then
                     return 1
                 else
                     return 0
                 end
             } 1 x
         ] 1
+        # error stats were not incremented
+        assert_equal [errorrstat ERR r] {}
+        assert_equal [errorrstat OOM r] {}
```

Comment:
don't we expect OOM to be incremented?

Reply:
Amm its a good point, what happened is that the OOM is raised as an error from `scriptCall` (no as a reply), so as long as the script do not propagate it we do not count it. Do you see a good way to fix it?

Reply:
Fixed